lib/request.js

```diff
@@ -0,0 +1,160 @@
+'use strict';
+
+var
+    https = require('https'),
+    {
+        KlaviyoApiError,
+        KlaviyoAuthenticationError,
+        KlaviyoRateLimitError,
+        KlaviyoServerError
+    } = require('./errors.js');
+
+const
+    KLAVIYO_API_HOSTNAME = 'a.klaviyo.com',
+    API_ROOT = 'api',
+    V1_API = 'v1',
+    V2_API = 'v2',
+    HTTPS_PORT = 443,
+    PACKAGE_VERSION = require('../package.json').version;
+
+
+class Request {
+    /**
+     * Make a request to one of the public API endpoints.
+     * @param {string} resource API resource to access.
+     * @param {string} data base64 encoded string.
+     * @returns {Promise.<number|KlaviyoApiError>} 1 (pass) or 0 (fail) | Non-200 response received.
+     */
+    static publicRequest ({
+        resource,
+        data
+    } = {}) {
+        const path = `/${API_ROOT}/${resource}?${data}`;
+        return _request({ path: path, method: 'GET' });
+    }
+
+    /**
+     * Make a request to a v1 API endpoint
+     * @param {string} resource API resource to access
+     * @param {string} method HTTP method
+     * @param {string} data URL encoded querystring
+     * @returns {Promise.<object|KlaviyoApiError>} Parsed JSON response from the API | Non-200 response received.
+     */
+    static v1Request({
+        resource,
+        method,
+        data=null
+    } = {}) {
+        const path = `/${API_ROOT}/${V1_API}/${resource}?${data}`;
+        const headers = {
+            'Content-Type': 'application/x-www-form-urlencoded'
+        };
+        return _request({ path: path, method: method, headers: headers });
```

Comment:
totally optional but you could use the object shorthand syntax here

```js
return _request({ path, method, headers });
```

Reply:
I am inclined to leave as-is just for consistency, happy to change it if we think it would make sense tho